java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -819,8 +822,11 @@ public ColumnView[] splitAsViews(int... indices) {
     ColumnView[] columnViews = new ColumnView[nativeHandles.length];
     try {
       for (int i = 0; i < nativeHandles.length; i++) {
-        columnViews[i] = new ColumnView(nativeHandles[i]);
+        long nativeHandle = nativeHandles[i];
+        // setting address to zero, so we don't clean it in case of an exception as it
+        // will be cleaned up by the constructor
         nativeHandles[i] = 0;
+        columnViews[i] = new ColumnView(nativeHandle);
```

Comment:
Same comment here. I think we want to move the try up to line 53 for this constructor.